common/src/main/java/org/schemarepo/validation/AllValidator.java

```diff
@@ -0,0 +1,44 @@
+package org.schemarepo.validation;
+import org.schemarepo.SchemaEntry;
+import org.schemarepo.SchemaValidationException;
+import org.schemarepo.Validator;
+import org.schemarepo.config.Config;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.Map;
+
+/**
+ * <p>
+ * A {@link Validator} for validating the provided schema against all
+ * schemas in the Iterable in {@link #validate(String, Iterable)}.
+ * </p>
+ * <p>
+ * Uses the {@link ValidationStrategy} provided in the constructor to
+ * validate the Schema against each Schema in the Iterable, in Iterator
+ * order, via {@link ValidationStrategy#validate(String, String)}.
+ * </p>
+ */
+public final class AllValidator implements Validator {

```

Comment:
Thanks for the javadoc. Very helpful.

Thanks to the doc, I think I get the intent of this class, but ```AllValidator``` seems a bit terse. Can we have a more self-explanatory name?

Perhaps ```ValidatorAgainstAllPreviousSchema```?

This is perhaps the other extreme, too long ;) , but I think it captures the intent, right?

Please feel free to suggest other names if you agree that ```AllValidator``` is too terse...